### PR TITLE
Take advantage of detached state for initializing model

### PR DESCRIPTION
**Pull request by ChumpChief**
Since FluidContainer supports detached state now, we can initialize the data prior to attaching and drop the initialize call which waits on the key to be ready.  This is recommendable and less error-prone (it's easy to forget that the second client might arrive before a post-attach value is set).  Here I do this via a static method on the controller.

This change also evicts some view logic from the app and restructures the view a bit -- probably this should just be converted to React at this point but I stop short of that here.

### CODE REVIEW

File: examples/hosts/app-integration/external-controller/src/controller.ts [hunk 1]
```diff
@@ -40,8 +40,20 @@ interface DiceRollerControllerProps {
...
+     */
+    public static initializeModel(props: DiceRollerControllerProps) {
+        props.set(diceValueKey, 1);
+    }
+
     constructor(private readonly props: DiceRollerControllerProps) {
         super();
+        const value = this.props.get(diceValueKey);
+        if (typeof value !== "number") {
+            throw new Error("Model is incorrect - did you call DiceRollerController.initialize() to set it up?");
```
Comment (skylerjokiel) [inline examples/hosts/app-integration/external-controller/src/controller.ts:55]:
```suggestion
            throw new Error("Model is incorrect - did you call DiceRollerController.initializeModel() to set it up?");
```

**Review by skylerjokiel — APPROVED** [review]
👌

**Review by micahgodbolt — APPROVED** [review]

File: examples/hosts/app-integration/external-controller/src/app.ts [hunk 2]
```diff
@@ -78,54 +88,36 @@ async function start(): Promise<void> {
     let id: string;
 
     // Get or create the document depending if we are running through the create new flow
-    const createNew = !location.hash;
+    const createNew = location.hash.length === 0;
```
Comment (micahgodbolt) [inline examples/hosts/app-integration/external-controller/src/app.ts:91]: maybe this should be <= 1

because `site.url/#` would return a hash length of 1

Comment (ChumpChief) [reply examples/hosts/app-integration/external-controller/src/app.ts:91]: Either one is fine -- `site.url/#` still gives a hash length of 0.

**Pull request merged**